src/main/kotlin/org/neo4j/graphql/CrudOperations.kt

```diff
@@ -34,6 +35,29 @@ fun augmentedSchema(ctx: Translator.Context, type: ObjectTypeDefinition): Augmen
     } else result
 }
 
+fun augmentedSchema(ctx: Translator.Context, source: ObjectTypeDefinition, target: ObjectTypeDefinition): Augmentation? {
+    val sourceIdField = source.fieldDefinitions.findID()
+    val targetIdField = target.fieldDefinitions.findID()
+    if (sourceIdField == null || targetIdField == null) {
+        return null
+    }
+    val sourceTypeName = source.name
+    val targetTypeName = target.name
+    return if (ctx.mutation.enabled && !ctx.mutation.exclude.contains(sourceTypeName)
```

Comment:
do this check first, before looking at id fields
then create an object that gets source and target passed in and has responsibility of creating the add and delete mutations

Reply:
so it can move a lot of the variables that we usually need to pass around into instance variables and all the code gets easier